beacon-chain/rpc/prysm/v1alpha1/validator/proposer_bellatrix.go

```diff
@@ -242,6 +242,14 @@ func (vs *Server) getPayloadHeaderFromBuilder(ctx context.Context, slot primitiv
 		return nil, fmt.Errorf("incorrect parent hash %#x != %#x", header.ParentHash(), h.BlockHash())
 	}
 
+	reg, err := vs.BeaconDB.RegistrationByValidatorID(ctx, idx)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not get validator registration")
```

Comment:
we should be more lenient here

Reply:
we register our validator ever slot now, but I'm not super confident yet, maybe there could be misses or timeouts that prevent us from getting from the cache